mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1402,6 +1547,32 @@ createCPUConvWithMLIR(ModuleOp module, func::FuncOp &func,
     break;
   }
 
+  Value opd1, opd2, result;
+
+  switch (genConfig.operation.value()) {
+  case rock::ConvOpType::Fwd:
```

Comment:
Split the block->getArgument parts of this switch out and ahead of the loop, so we can ensure they're F32 if necessary.